diffkemp/llvm_ir/build_llvm.py

```diff
@@ -124,6 +131,61 @@ def _get_kernel_tar_from_upstream(self):
 
         return tarname
 
+    def _get_kernel_tar_from_centos(self):
+        """
+        Download sources of the required kernel from the CentOS Git.
+        First the corresponding Git repository is downloaded and the URLs of the
+        necessary files are derived from it. Then it downloads them and extracts
+        the kernel source.
+        :returns Name of the tar file containing the sources.
+        """
+        centos_git_url = "https://git.centos.org/git/rpms/kernel.git"
+        self.configfile = "kernel-{}-x86_64.config".format(
+            self.kernel_version.split("-")[0])
+
+        tmpdir = mkdtemp()
+        cwd = os.getcwd()
+        os.chdir(tmpdir)
+
+        # Clone the Git repository and checkout the appropriate commit.
+        check_call(["git", "clone", "-q", centos_git_url])
+        os.chdir("kernel")
+        check_call(["git", "checkout", "-q",
+            "tags/imports/c7/kernel-{}.el7".format(self.kernel_version)])
+
+        # Get the hashes of the required files and the config from the metadata
+        # file
+        filelist = []
+        with open(".kernel.metadata", 'r') as metadata:
+            for line in metadata:
+                filelist.append(line.rstrip().split(' '))
+
+        shutil.copy("SOURCES/" + self.configfile, self.kernel_base_path)
+        os.chdir(self.kernel_base_path)
+        shutil.rmtree(tmpdir)
+
+        # Download the actual sources
+        for entry in filelist:
+            url = "https://git.centos.org/sources/kernel/c7/{}".format(entry[0])
+            filename = entry[1].split("/")[1]
+
+            urlretrieve(url, os.path.join(self.kernel_base_path, filename),
+                    show_progress)
+
+        # Delete all files except kernel sources tar, config file, and kabi
+        # whitelists tar if required (keep the directories since these are
+        # other kernels - RPM does not contain dirs).
+        tarname = "linux-{}.el7.tar.xz".format(self.kernel_version)
+        self.kabi_tarname = "kernel-abi-whitelists-{}.tar.bz2".format(
+            self.kernel_version.split("-")[1])
+        for f in os.listdir("."):
+            if (os.path.isfile(f) and f != tarname and f != self.configfile and
+                    f != self.kabi_tarname):
+                os.remove(f)
+        self.configfile = os.path.abspath(self.configfile)
+        os.chdir(cwd)
+        return tarname
```

Comment:
This code (from line 175) is essentially the same as in _get_kernel_tar_from_brew, it would be better to reuse it (put it in a separate function).